data/transactions/logic/eval.go

```diff
@@ -533,6 +533,28 @@ const (
 // StackTypes is an alias for a list of StackType with syntactic sugar
 type StackTypes []StackType
 
+func parseStackTypes(spec string) StackTypes {
+	if spec == "" {
+		return nil
+	}
+	types := make(StackTypes, len(spec))
+	for i, letter := range spec {
+		switch letter {
+		case 'a':
+			types[i] = StackAny
+		case 'b':
+			types[i] = StackBytes
+		case 'i':
+			types[i] = StackUint64
+		case 'x':
```

Comment:
@jannotti Optional:  Presumably it's illogical for `case 'x'` to be invoked when parsing `proto.Arg`.  Optionally, I'm wondering if you think it's worthwhile to rework `parseStackTypes` to panic?

I don't feel strongly.  I raise the thought because it may help isolate a problem faster.

Reply:
I don't really want to add a flag to parseStackTypes for that, because it seems like StackTypes is abstract enough that it ought not care which types it parses. I could see checking the returned type to ensure Arg does not contain StackNone.

I envision parseStackTypes getting more complex in the future, so that it contains more details (a byte array that must be len 32, for example), and possible names of the types to facilitate doc gen.  If so, it might make sense to separate argument and return lists, into different parse functions, and I'd certainly panic then.

Reply:
@jannotti Got it - Seems reasonable.  Thanks for sharing the context.  Resolving.